splunk_topology/check.py

```diff
@@ -105,7 +116,31 @@ def check(self, instance):
             self.start_snapshot(instance_key)
 
         try:
-            self._auth_session(instance)
+            if 'token' not in authentication:
+                self.log.debug("Using basic authentication mechanism")
+                self._auth_session(instance)
+            else:
+                self.log.debug("Using token based authentication mechanism")
+                persist_token_key = instance.instance_config.base_url + "token"
+                if self.status.data.get(persist_token_key) is None:
+                    self.log.debug("Creating a new token from first initial token")
+                    # Since this is first time when check starts with initial token,
+                    # we need to validate and create the new token
+                    if self._is_valid_token(instance, authentication["token"])[0]:
+                        new_token = self._create_auth_token(instance, authentication["token"])
+                        self.update_token_memory(instance.instance_config.base_url, new_token)
+                    else:
+                        msg = "Initial Token is expired, Please renew your token or use the valid token."
+                        raise Exception(msg)
+                token = self.status.data.get(persist_token_key)
+                token_flag, days_expiry = self._is_valid_token(instance, token)
+                if not token_flag:
+                    if days_expiry < 0:
```

Comment:
Quite redundant as `token_flag` is actually a boolean saying a token is valid or not.